oauthproxy.go

```diff
@@ -351,7 +366,7 @@ func buildSessionChain(opts *options.Options, sessionStore sessionsapi.SessionSt
 
 	if opts.SkipJwtBearerTokens {
```

Comment:
This function is a little bit problematic atm - as essentially the provider to use for cookie refreshes needs to be defined in the storedSessionLoaderOptions before the cookies are loaded (at which point i would like to pull the provider from the cookie session normally).

ideally the flow should be:

* Create session loader
* Extract provider ID, check validity, expiry etc
* find providerslice i that has this providerID
* implement ```RefreshSession:  opts.GetProvider(i).RefreshSession```

Reply:
Ideally the jwt bearer method should have similar flow implemented - probably also having a 

* Extract info from jwt (verifier-url/issuer url, groups, etc )
* lookup which provider from the verifier/issuer url is responsible for this token
* check allowed groups etc from the provider options
* create "session"

Reply:
I might add one comment to this, as we want LOAD functions to be as fast as possible, all this additional logic should only trigger once it is confirmed the cookie has expired